index.js

```diff
@@ -140,48 +140,66 @@ Pbf.prototype = {
 
     // verbose for performance reasons; doesn't affect gzipped size
 
-    readPackedVarint: function() {
-        var end = this.readVarint() + this.pos, arr = [];
+    readPackedVarint: function(arr) {
+        var isPacked = (this.buf[this.pos - 1] & 0x7) === Pbf.Bytes;
+        var end = isPacked ? this.readVarint() + this.pos : this.pos + 1;
```

Comment:
Can we cut this out into a method or a function? The same two lines are repeated 8 times... It would probably be inlined by V8 so hopefully not affecting performance.